lib/server.js

```diff
@@ -718,7 +718,7 @@ WalletService.prototype._getUtxosForAddresses = function(addresses, cb) {
       var u = _.pick(utxo, ['txid', 'vout', 'address', 'scriptPubKey', 'amount', 'satoshis', 'confirmations']);
       u.confirmations = u.confirmations || 0;
       u.locked = false;
-      u.satoshis = u.satoshis ? +u.satoshis : Utils.strip(u.amount * 1e8);
+      u.satoshis = _.isNumber(u.satoshis) ? +u.satoshis : Utils.strip(u.amount * 1e8);
```

Comment:
Maybe it's easier just change the calls to ``stubUtxos`` so there are no zero amounts passed? E.g. instead ``_.range(4)`` use ``_.range(1, 4)``. Anyway with this fix zero UTXOs do no harm.